packages/client/src/bags/types/bagOperation.types.ts

```diff
@@ -0,0 +1,177 @@
+  changeType: ChangeType.ItemIsUnavailable;
+  productId?: OperationChange['productId'];
+  variantId?: OperationChange['variantId'];
+  reason?: OperationChange['reason'];
+};
+
+type ChangeTypeItemMerged = {
+  changeType: ChangeType.ItemMerged;
+  productId?: OperationChange['productId'];
+  variantId?: OperationChange['variantId'];
+  oldValue?: OperationChange['oldValue'];
+  newValue?: OperationChange['newValue'];
+};
+
+type ChangeTypeItemOriginChanged = {
+  changeType: ChangeType.ItemOriginChanged;
+  productId?: OperationChange['productId'];
+  variantId?: OperationChange['variantId'];
+};
+
+type ChangeTypeItemPriceChanged = {
+  changeType: ChangeType.ItemPriceChanged;
+  itemId?: OperationChange['itemId'];
+  newValue?: OperationChange['newValue'];
+  oldValue?: OperationChange['oldValue'];
+  productId?: OperationChange['productId'];
+  variantId?: OperationChange['variantId'];
+};
+
+type ChangeTypeItemQuantityChanged = {
+  changeType: ChangeType.ItemQuantityChanged;
+  itemId?: OperationChange['itemId'];
+  newValue?: OperationChange['newValue'];
+  oldValue?: OperationChange['oldValue'];
+};
+
+type ChangeTypeItemNotEnoughStock = {
+  changeType: ChangeType.NotEnoughStock;
+  itemId?: OperationChange['itemId'];
+  stockAvailable?: OperationChange['stockAvailable'];
+};
+
+type ChangeTypePromoCodeEligibilityChanged = {
+  changeType: ChangeType.PromoCodeEligibilityChanged;
+};
+
+type ChangeTypePromotionNotApplied = {
+  changeType: ChangeType.PromotionNotApplied;
+};
+
+type ChangeTypeShippingCostChanged = {
+  changeType: ChangeType.ShippingCostChanged;
+};
+
+type ChangeTypeShippingFlatRateEligibilityChanged = {
+  changeType: ChangeType.ShippingFlatRateEligibilityChanged;
+};
+
+type ChangeTypeTotalDiscountChanged = {
+  changeType: ChangeType.TotalDiscountChanged;
+  newValue?: OperationChange['newValue'];
+  oldValue?: OperationChange['oldValue'];
+};
+
+type ChangeTypeSelectedSaleIntentChanged = {
+  changeType: ChangeType.SelectedSaleIntentChanged;
+  productId?: OperationChange['productId'];
+  variantId?: OperationChange['variantId'];
+  itemId?: OperationChange['itemId'];
+  oldValue?: OperationChange['oldValue'];
+  newValue?: OperationChange['newValue'];
+};
+
+export type BagOperationChange =
+  | ChangeTypeItemDiscountChanged
+  | ChangeTypeItemIsNowAvailable
+  | ChangeTypeItemIsUnavailable
+  | ChangeTypeItemMerged
+  | ChangeTypeItemOriginChanged
+  | ChangeTypeItemPriceChanged
+  | ChangeTypeItemQuantityChanged
+  | ChangeTypeItemNotEnoughStock
+  | ChangeTypePromoCodeEligibilityChanged
+  | ChangeTypePromotionNotApplied
+  | ChangeTypeSelectedSaleIntentChanged
+  | ChangeTypeShippingCostChanged
+  | ChangeTypeShippingFlatRateEligibilityChanged
+  | ChangeTypeTotalDiscountChanged;
+
+export enum BagViolationFixSuggestionAction {
+  DecreaseQuantity = 'DecreaseQuantity',
+  IncreaseValue = 'IncreaseValue',
+  DecreaseValue = 'DecreaseValue',
+}
+
+type BagViolationFixSuggestion = {
```

Comment:
This type should be exported as well since it is being used inside `BagViolation` type which is being exported.

Reply:
Done!